languages/csharp/generic/Parse_csharp_tree_sitter.ml

```diff
@@ -649,7 +678,7 @@ and interpolation_alignment_clause (env : env)
 
 and parenthesized_expression (env : env)
     ((_v1, v2, _v3) : CST.parenthesized_expression) =
-  expression env v2
+  non_lvalue_expression env v2
```

Comment:
it seems we do the same in Java, C, and C++. i'm not sure how we would factor the range of the parentheses in, honestly, since there's not space for it in an arbitrary `expr` variant. regardless, left a comment!